### DISCUSSION TITLE
Battery Health vs Range vs Charging %

### DISCUSSION THREAD

**Discussion by stauffenberg2020** [jheredianet/Teslamate-CustomGrafanaDashboards#86] [General]
Hi @jheredianet ,

Do you know if a dashboard can be made to show the Range vs charging % (Based on car's existing data)

And also in the same/different plot to show Battery health or Life vs the charging %
i.e,
If I always charge from 0% to 100%, then I would expect my battery life/health to be shorter, and range will be at it maximum
If I always charge from 20% to 80%, then I would expect my battery life/health to be decent, and range will be decent, And
If I always charge from 40% to 60%, then I would expect my battery life/health to be best, but my range will be pretty short

**Comment by jheredianet** [comment]
Hi, I'm sure it's possible, but not sure where to begin from now. 

Anyway, I don't see the use of that type of graph because reality is not like that, you never charge under a single pattern. I think that, on the contrary, it is recommended that the charges be variable and that they be in all ranges, not always the same ones. Experts say that batteries like to be in constant movement of electrons.

If you need for example 30% on a daily basis, the best thing is that one week you charge it from 20% to 50%, another from 30% to 60%, another 50 to 80%, etc., etc.
And from time to time, you do a long charge from as close to 0% to 100%. I don't think that the useful life of the battery should be calculated based on a specific pattern but rather the idea is to try to make it variable as I understand it.

I believe that the most accurate way to calculate or project battery life is by temperature and other conditions that have nothing to do with the percentage of charge in a specific range. 

I you see the Battery Health dashboad there is evolution graph wich is variable depending on each charge session which I believe is the real indicator of what the battery loses or gains depending on the charging conditions and the climate (or time of year)

**Reply by stauffenberg2020** [reply]
> If you need for example 30% on a daily basis, the best thing is that one week you charge it from 20% to 50%, another from 30% to 60%, another 50 to 80%, etc., etc. And from time to time, you do a long charge from as close to 0% to 100%. I don't think that the useful life of the battery should be calculated based on a specific pattern but rather the idea is to try to make it variable as I understand it.

I get it, The main reason I was thinking for this type of feature is, I currently rely on public AC charging and I do not have the possibility to have my own charger at my home. So, whenever I find the price is cheaper in the AC public chargers near me, I go and charge my car upto 90% and with this 90% battery, I will be able to use the car for nearly 10 days (6 days to work). I was worried if this high charge limit could affect my battery in the long run - Hence I wanted to see if there could be a direct correlation.

> If you see the Battery Health dashboad there is evolution graph which is variable depending on each charge session which I believe is the real indicator of what the battery loses or gains depending on the charging conditions and the climate (or time of year)

I understand that evolution of battery health over mileage plot. But as you said, it depends on both **charging conditions** and the **climate** (or time of the year). _Is it possible to distinguish them_? i.e Evolution of battery health only due to charging conditions in one plot and due to climate in another plot? Or if there is some other way to distinguish it

**Reply by jheredianet** [reply]
> I understand that evolution of battery health over mileage plot. But as you said, it depends on both **charging conditions** and the **climate** (or time of the year). _Is it possible to distinguish them_? i.e Evolution of battery health only due to charging conditions in one plot and due to climate in another plot? Or if there is some other way to distinguish it

It could be possible (hard because there is no a clear way to do it, maybe the data coming from Tesla should be different), the question is whether it is really worth making this distinction, because in the end, degradation is degradation, whatever the cause is.

**Reply by stauffenberg2020** [reply]
> > I understand that evolution of battery health over mileage plot. But as you said, it depends on both **charging conditions** and the **climate** (or time of the year). _Is it possible to distinguish them_? i.e Evolution of battery health only due to charging conditions in one plot and due to climate in another plot? Or if there is some other way to distinguish it
> 
> It could be possible (hard because there is no a clear way to do it, maybe the data coming from Tesla should be different), the question is whether it is really worth making this distinction, because in the end, degradation is degradation, whatever the cause is.

True, But degradation due to climate is not controllable - But degradation due to charging conditions is controllable

**Reply by jheredianet** [reply]
> True, but degradation due to climate is not controllable - But degradation due to charging conditions is controllable

"Degradation due to charging conditions is controllable", I'm not sure of that, at least I wouldn't know how to control that. Keep in mind that there is also degradation over time, even if you don't use the car.
In any case, with the "Charging Stats" dashboard you can get an idea of ​​the way in which the car is mainly charged, and therefore have a reference for the care of the battery.
